src/test_application/OSPMagnum.cpp

```diff
@@ -160,76 +162,178 @@ osp::active::ActiveScene& OSPMagnum::scene_create(std::string && name)
 void testapp::config_controls(OSPMagnum& rOspApp)
 {
     // Configure Controls
+    //Load toml
+    auto data = toml::parse("settings.toml");
+
+    osp::UserInputHandler& rUserInput = rOspApp.get_input_handler();
+    try {
+        for (const auto& [k, v] : data.as_table()) {
+            std::cout << k << std::endl;
+            std::string primary = toml::find(v, "primary").as_string();
+            std::vector<osp::ButtonVarConfig> controls = parse_control(primary);
+
+            std::string secondary = toml::find(v, "secondary").as_string();
+            std::vector<osp::ButtonVarConfig> secondaryKeys = parse_control(secondary);
+
+            controls.insert(controls.end(), secondaryKeys.begin(), secondaryKeys.end());
+
+            bool holdable = toml::find(v, "holdable").as_boolean();
+
+            //Add user input
+            rUserInput.config_register_control(k, holdable, controls);
+
+        }
+    }
+    catch (std::out_of_range oor) {
+        std::cout << oor.what() << std::endl;
+    }
+}
+
+
+//Map for all the keys
+//The tuple is in this order: device, number, and hold/pressed
+using Key_t = OSPMagnum::KeyEvent::Key;
+using Mouse_t = OSPMagnum::MouseEvent::Button;
+using VarOp_t = osp::ButtonVarConfig::VarOperator;
+using VarTrig_t = osp::ButtonVarConfig::VarTrigger;
+
+typedef std::tuple<int, int> button_tuple;
+const std::map<std::string_view, button_tuple, std::less<>> buttonMap = {
+    //Keyboard
+    {"LCtrl", {osp::sc_keyboard, (int)Key_t::LeftCtrl}},
```

Comment:
I'm sad that we have to have a manually specified list of strings like this.

but I can't think of any way to avoid it.

Can you?

Reply:
Nope, I think it's best to just suck it up in this case. It's ugly but doing it programmatically means circumventing Magnum's abstraction layer which is a bad idea imo.

Reply:
Noted.

Reply:
I did a programmatic method before this, but apparently not all devices map their button keys to ASCII, so this was the best  method. I didn't want to do this either.